### PR TITLE
golang: check for full discovery-go usage

### PR DESCRIPTION
**Pull request by nathanleiby**
We expect that discovery-go should be used to configure all dependencies
within the app (rather than directly looking them up by custom
environment variables)

Example output:

```
...

golang_checks.TestGolangSetup
        . test_check_golang_mk
        . test_drone_build_image
        F dependencies in launch yml must exactly match those called via discovery-go in the app: fake-service-aaa, fake-service-bbb

...

Report Card: 99/100 (A) ... Classic.
```

### CODE REVIEW

**Comment by nathanleiby** [comment]
@mohit FYI

**Comment by nathanleiby** [comment]
TODO: this doesn't yet have a way of thinking separately about gearman dependencies.

**Comment by nathanleiby** [comment]
@xavi- do you think this is worth merging as is, simply as a tool to check for discovery compliance? Should we hard-code the list of gearman apps, or is that something that should be private?

File: report_card/checks/golang_checks.py [hunk 1]
```diff
@@ -19,12 +22,29 @@
...
+
+def get_discovery_go_dep(line):
+    """ Given a line of code, extract the discovery-go dependency, if one is found
+    Example...
+      input:
+        main.go:\tif fooUrl, err := discovery.HostPort("foo-app", "thrift"); err != nil {\n
+      output:
+        foo-app
+    """
+    m = re.match('.*discovery\.(URL|HostPort|Host|Port)\("([a-zA-Z0-9_-]*)', line)
```
Comment (xavi-) [inline report_card/checks/golang_checks.py]: Nit.  Change to last part of regex to `"([a-zA-Z0-9_-]+)"`?  Uses `+` instead of `*` and has a close `"`.

**Comment by xavi-** [comment]
small nit.  lgtm

**Comment by mohit** [comment]
We should start thinking about checks that are very clever specific and pulling them as plugins. But for now, let's try to keep lists of things out :)

**Comment by nathanleiby** [comment]
@mohit agreed! https://clever.atlassian.net/browse/INFRA-1540

**Comment by nathanleiby** [comment]
This will report false positives for gearman deps for now, but I think it's a useful step forward nevertheless and we can refine once we have a plugin model

**Pull request merged**